framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -171,6 +173,37 @@ public Void visitTypeCast(TypeCastTree node, Void p) {
         if (node.getExpression().getKind() == Kind.NULL_LITERAL) {
             return null;
         }
+
+        AnnotatedTypeMirror castType = atypeFactory.getAnnotatedType(node);
+        AnnotationMirror castAnno = castType.getAnnotationInHierarchy(atypeFactory.UNKNOWNVAL);
+        AnnotationMirror exprAnno =
+                atypeFactory
+                        .getAnnotatedType(node.getExpression())
+                        .getAnnotationInHierarchy(atypeFactory.UNKNOWNVAL);
+
+        // Don't check a type cast if it is to a declared IntRange that covers the whole
```

Comment:
This sentence is a bit confusing.  For example, what does it mean to "cover the whole type"?   What about something like:
It is always legal to cast a type to a type that includes all possible values for that type.